p2p/src/net/default_backend/peer.rs

```diff
@@ -121,28 +125,49 @@ where
             peer_event_tx,
             backend_event_rx,
             node_protocol_version,
+            time_getter,
         }
     }
 
     fn validate_peer_time(
         p2p_config: &P2pConfig,
-        local_time: Duration,
-        remote_time: Duration,
+        local_time: std::ops::RangeInclusive<Time>,
+        remote_time: P2pTimestamp,
     ) -> crate::Result<()> {
         // TODO: If the node's clock is wrong and we disconnect peers,
         // it can be trivial to isolate the node by connecting malicious nodes
         // with the same invalid clock (while honest nodes can't connect).
         // After that, the node is open to all kinds of attacks.
-        let time_diff =
-            std::cmp::max(local_time, remote_time) - std::cmp::min(local_time, remote_time);
+
+        // We do not know at what point exactly the peer recorded its timestamp. However, we do
+        // know it was somewhere between the request was initiated and the response was received.
+        // We give the peer some leeway when it comes to network latency so the acceptable time is
+        // in the interval [`init_time - tolerance`, `recv_time + tolerance`].
+        //
+        // Since the distance between `init_time` and `recv_time` is bounded by the handshake
+        // timeout, the effective max peer time offset is:
+        // ```tolerance + (recv_time - init_time) <= tolerance + handshake_timeout```
```

Comment:
I'm a bit confused by this. Did you mean `2*tolerance + (recv_time - init_time)`? Because that's the length of the interval that you've mentioned.

Reply:
To summarise, the effective tolerances are:
* `tolerance + handshake_timeout` between our node and any one given peer
* `2 * tolerance + hendshake_timeout` between any two peer the node is connected to

Added the note of this to the comment, hopefully it's now clearer.